src/backend/distributed/master/master_stage_protocol.c

```diff
@@ -544,18 +551,14 @@ CreateShardsOnWorkers(Oid distributedRelationId, List *shardPlacements,
 		 */
 		if (ShouldRecordRelationAccess() && useExclusiveConnection && partitionTable)
 		{
-			RelationShard *parentRelationShard = CitusMakeNode(RelationShard);
```

Comment:
I think before this change we were missing few relationShards, especially it is not a partitioned table: (a) the placement itself (b) fkey relationships. This PR makes sure to cover those as well.